Chono/AnnotationSource.cs

```diff
@@ -57,6 +63,32 @@ public AnnotationSource(IPluginHost host, IProject project)
             StringBuilder bldr = new StringBuilder();
             int currentLevelInBuilder = -1;
 
+            void AppendQuotationMark(string mark)
+            {
+	            if (mark.Length > 1)
+		            bldr.Append("(");
+	            bldr.Append(Regex.Escape(mark));
+	            if (mark.Length > 1)
+		            bldr.Append(")");
+            }
+            
+            void AppendNewLevel(int level, string mark)
+            {
+	            bldr.Append("(?<");
+	            bldr.Append(kRgxOpenLevelGroupPrefix);
+	            bldr.Append(level);
+	            allMarks[mark] = bldr.Length;
+	            bldr.Append(">");
+            }
+            
+            void Insert(int i, int level, string mark)
+            {
+	            bldr.Insert(i, level);
+	            var keysToIncrement = allMarks.Where(kvp => kvp.Value >= allMarks[mark])
+		            .Select(kvp => kvp.Key).ToHashSet();
```

Comment:
Why do you make it a HashSet just so you can loop through it? Why not use ``.Distinct`` instead?